tracing-test-utils/src/main/java/com/palantir/tracing/TestTracingExtension.java

```diff
@@ -32,18 +32,23 @@
 import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
 import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
 import org.junit.platform.commons.support.AnnotationSupport;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 final class TestTracingExtension implements BeforeTestExecutionCallback, AfterTestExecutionCallback {
 
     private static final Logger log = LoggerFactory.getLogger(TestTracingExtension.class);
-    private final TestTracingSubscriber subscriber = new TestTracingSubscriber();
+    private static final Namespace TRACING_EXT_NAMESPACE = Namespace.create(TestTracingExtension.class);
+    private static final String SUBSCRIBER_STORE_KEY = "subscriber";
 
     @Override
     public void beforeTestExecution(ExtensionContext context) {
         Tracer.setSampler(AlwaysSampler.INSTANCE);
+        TestTracingSubscriber subscriber = context.getStore(TRACING_EXT_NAMESPACE)
+                .getOrComputeIfAbsent(
+                        SUBSCRIBER_STORE_KEY, _k -> new TestTracingSubscriber(), TestTracingSubscriber.class);
         Tracer.subscribe(context.getUniqueId(), subscriber);
 
         // TODO(dfox): sample can be modified by other code, we should be try ensure that the trace is always sampled
```

Comment:
@iamdanfox Do you happen to remember what these 2 TODOs are? Seems like this PR resolves them?

Reply:
The only TODO that is resolved by this PR is `TODO(dfox): clear existing tracing??`. The other ones are still relevant.

Although there not really anything we can do about `TODO(dfox): sample can be modified by other code`, so we may just want to remove it.